### ISSUE TITLE
Building a game project issues a "Unable to set EXE name/description" error

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2376]
**Describe the bug**
Under certain conditions building a game in the Editor issues a "Unable to set EXE name/description: LoadLibrary failed" error. This likely is related to Editor changing executable's meta-info (like author, release date etc).
I noticed that this happens 100% time right after a new sprite was imported to the project, although I cannot tell how that may be related.
"Attach game data to EXE" setting is disabled, and "Sprite compression" set to "None", if that matters.

**AGS Version**
3.6.1.22, does **not** happen in earlier major version 3.6.0.

**To Reproduce**
1. Launch the Editor, load any valid game project.
2. Do Build->Build EXE. Notice there are no errors.
3. Import 1 random sprite.
4. Do Build->Build EXE. Observe the error.

**Expected behavior**
Editor should not fail to update executable's info because a sprite was imported...

**Comment by ericoporto** [comment]
Is there a way to get a more meaningful message from `LoadLibrary` and why it failed?

**Comment by ericoporto** [comment]
Made a change that will make `LoadLibrary` show a better error message in the editor.

https://github.com/ericoporto/ags/commit/4b2563626f86d4a94eeb288cfbe722357c0609e1

Note: I can't reproduce this issue you have.

**Comment by ivan-mogilko** [comment]
The error returned is "Error #2, The system cannot find the file specified."

**Comment by ericoporto** [comment]
I don't remember where but I think we modify how LoadLibrary sees things, I think it uses the Editor path as reference while File Exists and other calls are using whatever is the current directory (the game project dir usually). So non-absolute paths work weird.

**Comment by ericoporto** [comment]
Not entirely sure, but I think LoadLibrary uses what is set in SetDllDirectory and File.Exists uses Current Working Dir, as reference for relative paths.

**Comment by ivan-mogilko** [comment]
Alright, two things. Firstly, using absolute path seems to fix this.

Then, my curiosity was too strong, because I wanted to know how is this related to sprites.
So I tried to narrow it down by disabling various parts of sprite import, and found the exact piece of code that was triggering this problem:
https://github.com/adventuregamestudio/ags/blob/bda4fe661f76da90ec148fbe8f5ea64797918443/Editor/AGS.Editor/GIFLoader/GifDecoder.cs#L14

Basically, calling `new MagickImageInfo(fileName)` in 3.6.1 does something that prevents our LoadLibrary to work as before. Maybe it's doing this SetDllDirectory unknown to us (since you mentioned that), but I may only guess.

**Comment by ericoporto** [comment]
I kept the same version number, but I switched the Magick package to one that works with both x64 and x86 systems. So perhaps I broke this but I didn't notice. This was made to be easy to build the Editor to x64 or x86. It looks like the way they make it work is by using SetDllDirectory...

https://github.com/dlemstra/Magick.NET/commit/52c8db309067695a6b94c6291020ac0096b6585e

I only know about this because AGS wasn't working for me through python in agstoolbox because PyInstaller uses SetDllDirectory internally, I devised a "lock/unlock" mechanism here: https://github.com/ericoporto/agstoolbox/commit/d2e887a5d8ec4ed14a61af195ecce6ccdc6f67a2 (to fix https://github.com/ericoporto/agstoolbox/issues/31)

**Comment by ivan-mogilko** [comment]
> I think we can just use GetDllDirectoryA, store what it is, use SetDllDirectory with NULL and then reset it if it was anything non-null.

That might have been another option... Maybe it's worth doing that too.
For this time I went with absolute paths though... and then I found that more calls require absolute paths (BeginUpdateResource). And then absolute paths broke things for games with unicode paths (e.g. in "Users/Username").
So in the end I also had to replace with widestring variants of few functions.
Had to test game project in non-latin unicode path for this.

Full list of commits:
a762e6bff01bb5b64495871024dc983e3fa5eb6d
db9f2789a5bbd71921e08f2891c9a680b447ccdf
bfdeca0d1c9c96ac0c2ee8054ae7d76166b3b31e
92e2c870fc1ce1c48051f3433e960537058e6fc1
20dbf113d7e478c3b719b3a1cc3d090948818185 -- this one was not entirely necessary, but I noticed last place where AGS.Native uses fopen and FILE instead of our streams.
